src/sentry/tsdb/redis.py

```diff
@@ -817,9 +883,17 @@ def merge_frequencies(self, model, destination, sources, timestamp=None, environ
                             ]
                         )
 
+            try:
+                responses = cluster.execute_commands(exports)
+            except Exception:
+                if durable:
+                    raise
+                else:
+                    continue
```

Comment:
This control flow is pretty gross, but I wanted to minimize the scope of this change and this was the least invasive way I could think to do so.